### DISCUSSION TITLE
Using pydantic.v1 models in pydantic (v2) models.

### DISCUSSION THREAD

**Discussion by jakub-mikians-airspace-intelligence** [pydantic/pydantic#8252] [Question]
I have a large codebase with many libraries in different repositories, many of them using pydantic v1. I want to _gradually_ migrate the libraries to support both pydantic 1 and pydantic 2 (when possible). In particular, I use `pydantic.v1` in the libraries.

Now I have a problem with a new package using such a library. The minimal (not)working example is below. F1 which is a "v1" base model cannot be used in F2 which is a "v2" base model.

Is there any way to make those two base models work?

In my case, due to large code base, migrating everything to pydantic v2 at once is not an option. Such a migration would block the development for significant time until all the packages would be migrated. 

```python
from pydantic.v1 import BaseModel as BaseModel1
from pydantic import BaseModel as BaseModel2


class F1(BaseModel1):
    pass

class F2(BaseModel2):
    f1: F1

F2(f1=F1())
```

```bash
Traceback (most recent call last):
  File "/Users/mikiansj/Downloads/compat.py", line 11, in <module>
    F2(f1=F1())
  File "/opt/homebrew/lib/python3.11/site-packages/pydantic/main.py", line 165, in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)
TypeError: BaseModel.validate() takes 2 positional arguments but 3 were given
```

**Comment by nachonavarro** [comment]
Has anybody figured out how to solve this issue?
